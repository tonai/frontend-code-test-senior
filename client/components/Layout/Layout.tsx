import { ReactNode } from "react";

import CartProvider from "../CartProvider/CartProvider";
import Header from "../Header/Header";
import Footer from "../Footer/Footer";

export interface ILayoutProps {
  children: ReactNode;
}

export default function Layout(props: ILayoutProps) {
  const { children } = props;
  return (
    <CartProvider>
      <Header />
      {children}
      <Footer />
    </CartProvider>
  );
}
