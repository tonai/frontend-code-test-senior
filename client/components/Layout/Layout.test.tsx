import { render } from "@testing-library/react";

import Layout from "./Layout";

describe("Layout component", () => {
  it("matches snapshot", () => {
    const { container } = render(<Layout>test</Layout>);
    expect(container).toMatchSnapshot();
  });
});
