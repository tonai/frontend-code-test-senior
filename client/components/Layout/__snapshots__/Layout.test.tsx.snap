// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component matches snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <img
      alt="Octopus Energy Logo"
      data-nimg="1"
      decoding="async"
      height="27"
      loading="lazy"
      src="/octopus-logo.svg"
      style="color: transparent;"
      width="192"
    />
    <div
      class="cart"
    >
      <img
        alt="Basket"
        data-nimg="1"
        decoding="async"
        height="27"
        loading="lazy"
        src="/basket.svg"
        style="color: transparent;"
        width="30"
      />
    </div>
  </header>
  test
  <footer
    class="footer"
  >
    <p>
      Octopus Energy Ltd is a company registered in England and Wales.
    </p>
    <p>
      Registered number: 09263424. Registered office: 33 Holborn, London, EC1N 2HT. Trading Office: 20-24 Broadwick Street, London, W1F 8HT
    </p>
  </footer>
</div>
`;
